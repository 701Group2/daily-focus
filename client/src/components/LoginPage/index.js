import { Button, Grid, Link, makeStyles, Paper, TextField } from "@material-ui/core";
import { useState } from "react";
import { ReactComponent as FocusLogo } from "./logo.svg";

const useStyles = makeStyles((theme) => ({
    container: {
        display: "flex",
        flexDirection: "column",
        height: "100vh",
        justifyContent: "center",
    },
    login: {
        margin: "auto",
        padding: theme.spacing(3),
        width: 440,
    },
    signInForm: {
        width: "100%",
    },
    signInTextField: {
        margin: "8px 0px",
        width: "100%",
    },
    signInBtn: {
        backgroundColor: "#30a0f5",
        borderRadius: "28px",
        color: "#ffffff",
        height: "56px",
        margin: "8px 0 16px 0",
        width: "100%",
    },
    signUpRedirect: {
        margin: "16px 0",
    },
}));

const LoginPage = () => {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");

    const classes = useStyles();

    const fetchLogin = (username, password) => {
        fetch("http://localhost:9000/login", {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify({
                email: username,
                password: password,
            }),
            referrerPolicy: "no-referrer",
        }).then((response) => {
            if (!response.ok) {
                // show user error message for incorrect user/password
            } else {
                const data = response.json();
                console.log(data.token);
                // retrieve user token
                // persist user token with use-persisted state
            }
        });
    };

    const submitLogin = (e) => {
        e.preventDefault();
        console.log(email);
        console.log(password);
        fetchLogin(email, password);
    };

    return (
        <div className={classes.container}>
            <Paper className={classes.login}>
                <Grid container direction="column" justify="center" alignItems="center" spacing={0}>
                    <Grid item>
                        <div>
                            <FocusLogo width={380} fill="#454545" />
                        </div>
                    </Grid>
                    <form className={classes.signInForm} onSubmit={submitLogin}>
                        <Grid item>
                            <TextField
                                className={classes.signInTextField}
                                label="Email address"
                                variant="outlined"
                                value={email}
                                onInput={(e) => setEmail(e.target.value)}
                            />
                        </Grid>
                        <Grid item>
                            <TextField
                                className={classes.signInTextField}
                                label="Password"
                                type="password"
                                variant="outlined"
                                value={password}
                                onInput={(e) => setPassword(e.target.value)}
                            />
                        </Grid>
                        <Grid item>
                            <Button className={classes.signInBtn} type="submit" variant="contained">
                                Sign In
                            </Button>
                        </Grid>
                    </form>
                    <Grid item className={classes.signUpRedirect}>
                        <Link href="#">Don't have an account? Sign up</Link>
                    </Grid>
                </Grid>
            </Paper>
        </div>
    );
};

export default LoginPage;
