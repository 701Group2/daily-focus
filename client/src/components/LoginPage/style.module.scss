.container {
    display: flex;
    height: 100vh;
    justify-content: center;
    margin: auto;
    min-width: 440px;
    width: 50vw;

    .logInContainer {
        display: flex;
        flex-direction: column;
        align-items: center;

        a {
            color: #30a0f5;
            font-size: 1.25rem;
            text-decoration: none;
        }
    }
}
