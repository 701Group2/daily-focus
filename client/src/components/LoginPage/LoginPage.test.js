import React from "react";
import { shallow, mount } from "enzyme";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import ShallowRenderer from "react-test-renderer/shallow";

import LoginPage from ".";

let component;

beforeEach(() => {
    component = shallow(<LoginPage />);
});

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<LoginPage />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("Route changes to '/path' when sign up clicked", () => {
    const history = createMemoryHistory();
    const wrapper = mount(
        <Router history={history}>
            <LoginPage />
        </Router>
    );
    wrapper.find("a").simulate("click", { button: 0 });
    expect(history.location.pathname).toBe("/");
});
