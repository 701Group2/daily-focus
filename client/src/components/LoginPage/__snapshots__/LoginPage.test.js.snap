// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="container"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-2"
  >
    <div
      className="logInContainer"
    >
      <div
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <ForwardRef(SvgFocusLogoWAccent)
          fill="#454545"
          width={380}
        />
      </div>
      <form
        onSubmit={[Function]}
      >
        <WithStyles(ForwardRef(TextField))
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          label="Email address"
          onInput={[Function]}
          style={
            Object {
              "margin": "10px 0px",
              "width": "100%",
            }
          }
          value=""
          variant="outlined"
        />
        <WithStyles(ForwardRef(TextField))
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          label="Password"
          onInput={[Function]}
          style={
            Object {
              "margin": "10px 0px",
              "width": "100%",
            }
          }
          type="password"
          value=""
          variant="outlined"
        />
        <WithStyles(ForwardRef(Button))
          className="makeStyles-button-3"
          color="primary"
          type="submit"
          variant="contained"
        >
          Sign In
        </WithStyles(ForwardRef(Button))>
      </form>
      <div
        style={
          Object {
            "padding": "32px",
          }
        }
      >
        <Link
          to="/"
        >
          Don't have an account? Sign up
        </Link>
      </div>
    </div>
  </WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Snackbar))
    action={
      <React.Fragment>
        <WithStyles(ForwardRef(Button))
          color="inherit"
          onClick={[Function]}
          size="small"
        >
          CLOSE
        </WithStyles(ForwardRef(Button))>
      </React.Fragment>
    }
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    autoHideDuration={5000}
    message="Username and/or password is incorrect"
    onClose={[Function]}
    open={false}
  />
</div>
`;
